models/job.js

```diff
@@ -18,6 +18,11 @@ const MODEL = {
 
     permutations: validator.jobPermutations,
 
+    prNum: Joi
```

Comment:
If a job is not a PR job, `prNum` may be `undefined` but it should cause validation error.

Reply:
`optional()` is not necessarily needed.
> Marks a key as optional which will allow undefined as values. Used to annotate the schema for readability as all keys are optional by default.

https://github.com/hapijs/joi/blob/v14.3.0/API.md#anyoptional

Reply:
I see.
I concerned about cases of `prNum: undefined` or `prNum: null` pattern. It's same as `configPipelineId` on 
https://github.com/screwdriver-cd/data-schema/blob/master/models/pipeline.js#L60-L62

Reply:
We should use `default(null)`?

Reply:
Default `optional()` might work.

Reply:
I deleted `prNum` for this time.
I will be added in Second Pass. 
https://github.com/screwdriver-cd/screwdriver/issues/1010#issuecomment-441493555